### PR TITLE
Network: Use CanonicalNetworkAddressFromAddressAndPort rather than manual fmt.Sprintf

### PR DESCRIPTION
**Pull request by tomponline**
- Further improves upon https://github.com/lxc/lxd/pull/7796/commits/6f3bd86122b84ef64b5cc2c4c6662a8afe942aee by using the canonical address (including port) for comparison rather than just the requested listen IP, as in the default case the existing listen socket was not detected and failed `lxd init`.
- Uses `CanonicalNetworkAddressFromAddressAndPort` rather than manual `fmt.Sprintf`, to make explicit which address types we accept.
- Converts `shared.DefaultPort` to an int, and replaces some internal usage of hardcoded "8443" with it.

### CODE REVIEW

**Comment by tomponline** [comment]
jenkins: test this please

File: lxd/main_init_interactive.go [hunk 1]
```diff
@@ -127,7 +127,7 @@ func (c *cmdInit) askClustering(config *cmdInitData, d lxd.InstanceServer) error
 
 			s, _, err := d.GetServer()
 			if err == nil {
-				if s.Config["cluster.https_address"] == value || s.Config["core.https_address"] == value {
+				if s.Config["cluster.https_address"] == address || s.Config["core.https_address"] == address {
```
Comment (stgraber) [inline lxd/main_init_interactive.go:130]: Why this change?

As far as I know the value stored in the database is the value as provided by the user, not the value after processing by CanonicalNetworkAddress so we should be checking `value` against the current value and not the updated `address`.

At least that's why I chose to go with value instead of address when I added this last week :)

Comment (tomponline) [reply lxd/main_init_interactive.go:130]: @stgraber I looked into this as was still getting listen errors on subsequent lxd inits after initial lxd init failed part way through for some unrelated reason. 

I found that the output of canonical address is stored in db here https://github.com/tomponline/lxd/blob/5fe4ae42c0c3fd520e2daf31165b6eea8c955157/lxd/main_init_interactive.go#L146 see line above that too.

Comment (stgraber) [reply lxd/main_init_interactive.go:130]: Oh, fun, I checked the server side code, not the client side code :)
